diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerModule.kt
@@ -4,7 +4,6 @@ import android.Manifest
 import android.content.Context
 import android.content.Intent
 import android.content.pm.PackageManager
-import android.net.Uri
 import android.os.Build
 import android.os.Bundle
 import android.os.OperationCanceledException
@@ -17,6 +16,7 @@ import expo.modules.imagepicker.contracts.CropImageContractOptions
 import expo.modules.imagepicker.contracts.ImageLibraryContract
 import expo.modules.imagepicker.contracts.ImageLibraryContractOptions
 import expo.modules.imagepicker.contracts.ImagePickerContractResult
+import expo.modules.imagepicker.contracts.MediaList
 import expo.modules.interfaces.permissions.Permissions
 import expo.modules.interfaces.permissions.PermissionsResponse
 import expo.modules.interfaces.permissions.PermissionsResponseListener
@@ -302,6 +302,6 @@ class ImagePickerModule : Module() {
  * Simple data structure that would be preserved after the Activity is destroyed.
  */
 internal data class PendingMediaPickingResult(
-  val data: List<Pair<MediaType, Uri>>,
+  val data: MediaList,
   val options: ImagePickerOptions
 )
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerResponse.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerResponse.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerResponse.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerResponse.kt
@@ -7,7 +7,7 @@ import expo.modules.kotlin.types.Enumerable
 
 internal class ImagePickerAsset(
   @Field val assetId: String? = null,
-  @Field val type: MediaType = MediaType.IMAGE,
+  @Field val type: MediaType? = MediaType.IMAGE,
   @Field val uri: String = "",
   @Field val width: Int = 0,
   @Field val height: Int = 0,
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerUtils.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerUtils.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerUtils.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/ImagePickerUtils.kt
@@ -20,6 +20,7 @@ import java.io.File
 import java.io.FileNotFoundException
 import java.io.FileOutputStream
 import java.io.IOException
+import kotlin.io.use
 
 internal fun createOutputFile(cacheDir: File, extension: String): File {
   val filePath = FileUtilities.generateOutputPath(cacheDir, ImagePickerConstants.CACHE_DIR_NAME, extension)
@@ -30,10 +31,30 @@ internal fun createOutputFile(cacheDir: File, uri: Uri): String =
   }
 }
 
-internal fun getType(contentResolver: ContentResolver, uri: Uri): String =
-  contentResolver.getType(uri)
+internal fun getType(contentResolver: ContentResolver, uri: Uri): String? {
+  val mimeFromCursor: () -> String? = {
+    contentResolver
+      .query(
+        uri,
+        listOf(DocumentsContract.Document.COLUMN_MIME_TYPE).toTypedArray(),
+        null,
+        null,
+        null
+      )?.use { cursor ->
+        if (cursor.moveToFirst()) {
+          val columnIndex = cursor.getColumnIndex(DocumentsContract.Document.COLUMN_MIME_TYPE)
+          if (columnIndex != -1 && !cursor.isNull(columnIndex)) {
+            return@use cursor.getString(columnIndex)
+          }
+        }
+        return@use null
+      }
+  }
+
+  return contentResolver.getType(uri)
+    ?: mimeFromCursor()
+    ?: getTypeFromFileUrl(uri.toString())
+}
 
 private fun getTypeFromFileUrl(url: String): String? {
   val extension = MimeTypeMap.getFileExtensionFromUrl(url)
@@ -76,12 +97,12 @@ internal fun String.toImageFileExtension(): String = when {
   else -> ".jpeg"
 }
 
-internal fun Uri.toMediaType(contentResolver: ContentResolver): MediaType {
+internal fun Uri.toMediaType(contentResolver: ContentResolver): MediaType? {
   val type = getType(contentResolver, this)
   return when {
-    type.contains("image/") -> MediaType.IMAGE
-    type.contains("video/") -> MediaType.VIDEO
-    else -> throw FailedToDeduceTypeException()
+    type?.contains("image/") == true -> MediaType.IMAGE
+    type?.contains("video/") == true -> MediaType.VIDEO
+    else -> null
+  }
+}
+
+private fun getTypeFromFileUrl(url: String): String? {
+  val extension = MimeTypeMap.getFileExtensionFromUrl(url)
+  return extension?.let { MimeTypeMap.getSingleton().getMimeTypeFromExtension(it) }
+}
+
+internal fun String.toImageFileExtension(): String = when {
+  this.endsWith("png", ignoreCase = true) -> ".png"
+  this.endsWith("gif", ignoreCase = true) -> ".gif"
+  this.endsWith("bmp", ignoreCase = true) -> ".bmp"
+  this.endsWith("webp", ignoreCase = true) -> ".webp"
+  !this.endsWith("jpeg", ignoreCase = true) -> {
+    Log.w(TAG, "Image file $this is of unsupported type. Falling back to JPEG instead.")
+    ".jpeg"
   }
+  else -> ".jpeg"
 }
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/MediaHandler.kt
@@ -6,6 +6,7 @@ import android.net.Uri
 import android.provider.OpenableColumns
 import android.util.Base64
 import androidx.core.net.toUri
+import expo.modules.imagepicker.contracts.MediaList
 import expo.modules.imagepicker.exporters.CompressionImageExporter
 import expo.modules.imagepicker.exporters.ImageExporter
 import expo.modules.imagepicker.exporters.RawImageExporter
@@ -19,13 +20,17 @@ internal class MediaHandler(
     get() = requireNotNull(appContextProvider.appContext.reactContext) { "React Application Context is null" }
 
   internal suspend fun readExtras(
-    bareResult: List<Pair<MediaType, Uri>>,
+    bareResult: MediaList,
     options: ImagePickerOptions
   ): ImagePickerResponse {
     val results = bareResult.map { (mediaType, uri) ->
       when (mediaType) {
         MediaType.VIDEO -> handleVideo(uri)
         MediaType.IMAGE -> handleImage(uri, options)
+        null -> ImagePickerAsset(
+          type = null,
+          uri = uri.toString()
+        )
       }
     }
 
@@ -47,7 +52,7 @@ internal class MediaHandler(
     } else {
       CompressionImageExporter(appContextProvider, options.quality)
     }
-    val mimeType = getType(context.contentResolver, sourceUri)
+    val mimeType = requireNotNull(getType(context.contentResolver, sourceUri))
     val outputFile = createOutputFile(cacheDirectory, mimeType.toImageFileExtension())
 
     val exportedImage = exporter.exportAsync(sourceUri, outputFile, context.contentResolver)
@@ -73,17 +78,21 @@ internal class MediaHandler(
     )
   }
 
-  private fun getAdditionalFileData(uri: Uri): AdditionalFileData? = context.contentResolver.query(uri, null, null, null, null)?.use { cursor ->
-    val nameIndex = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME)
-    val sizeIndex = cursor.getColumnIndex(OpenableColumns.SIZE)
-    cursor.moveToFirst()
-
-    val name: String? = cursor.getString(nameIndex)
-    val size = cursor.getLong(sizeIndex)
-    AdditionalFileData(
-      name,
-      size
-    )
+  private fun getAdditionalFileData(uri: Uri): AdditionalFileData? {
+    val queriedFields = listOf(OpenableColumns.DISPLAY_NAME, OpenableColumns.SIZE).toTypedArray()
+    return context.contentResolver.query(uri, queriedFields, null, null, null)?.use { cursor ->
+      if (!cursor.moveToFirst()) {
+        return null
+      }
+      runCatching {
+        val name = cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME).takeIf { it != -1 }?.let { cursor.getString(it) }
+        val size = cursor.getColumnIndex(OpenableColumns.SIZE).takeIf { it != -1 }?.let { cursor.getLong(it) }
+        AdditionalFileData(
+          name,
+          size
+        )
+      }.getOrNull()
+    }
   }
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ContractsUtils.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ContractsUtils.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ContractsUtils.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ContractsUtils.kt
@@ -3,11 +3,13 @@ import android.net.Uri
 import expo.modules.imagepicker.MediaType
 
+typealias MediaList = List<Pair<MediaType?, Uri>>
+
 /**
  * Data required to be returned upon successful contract completion
  */
-internal sealed class ImagePickerContractResult private constructor() {
-  class Success(val data: List<Pair<MediaType, Uri>>) : ImagePickerContractResult()
+internal sealed class ImagePickerContractResult {
+  class Success(val data: MediaList) : ImagePickerContractResult()
   object Cancelled : ImagePickerContractResult()
   object Error : ImagePickerContractResult()
 }
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CropImageContract.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CropImageContract.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CropImageContract.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/CropImageContract.kt
@@ -3,6 +3,7 @@ import android.app.Activity
 import android.content.Context
 import android.content.Intent
+import android.graphics.Bitmap
 import android.os.Build
 import androidx.core.net.toFile
 import androidx.core.net.toUri
@@ -27,7 +28,7 @@ internal class CropImageContract(
 ) : AppContextActivityResultContract<ImageLibraryContractOptions, ImagePickerContractResult> {
   override fun createIntent(context: Context, input: CropImageContractOptions) = Intent(context, expo.modules.imagepicker.ExpoCropImageActivity::class.java).apply {
     val mediaType = expo.modules.imagepicker.getType(context.contentResolver, input.sourceUri.toUri())
-    val compressFormat = mediaType.toBitmapCompressFormat()
+    val compressFormat = mediaType?.toBitmapCompressFormat() ?: Bitmap.CompressFormat.JPEG
     val cacheDirectory = appContextProvider.appContext.cacheDirectory
     val outputUri = createOutputFile(cacheDirectory, compressFormat.toImageFileExtension()).toUri()
 
diff --git a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ImageLibraryContract.kt b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ImageLibraryContract.kt
--- a/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ImageLibraryContract.kt
+++ b/node_modules/expo-image-picker/android/src/main/java/expo/modules/imagepicker/contracts/ImageLibraryContract.kt
@@ -1,8 +1,9 @@
 package expo.modules.imagepicker.contracts
 
 import android.app.Activity
 import android.content.ContentResolver
 import android.content.Context
 import android.content.Intent
 import android.net.Uri
+import android.util.Log
@@ -76,38 +77,46 @@ internal class ImageLibraryContract(
   }
 
   override fun parseResult(input: ImageLibraryContractOptions, resultCode: Int, intent: Intent?) =
-    if (resultCode == Activity.RESULT_CANCELED) {
-      ImagePickerContractResult.Cancelled
-    } else {
-      intent?.takeIf { resultCode == Activity.RESULT_OK }?.getAllDataUris()?.let { uris ->
-        if (input.options.allowsMultipleSelection) {
-          val results = uris.map { uri ->
-            uri.toMediaType(contentResolver) to uri
-          }.let {
-            if (input.options.selectionLimit > 0) {
-              it.take(input.options.selectionLimit)
+    try {
+      if (resultCode == Activity.RESULT_CANCELED) {
+        ImagePickerContractResult.Cancelled
+      } else {
+        intent?.takeIf { resultCode == Activity.RESULT_OK }?.getAllDataUris()?.let { uris ->
+          if (input.options.allowsMultipleSelection) {
+            val results = uris.map { uri ->
+              uri.toMediaType(contentResolver) to uri
+            }.let {
+              if (input.options.selectionLimit > 0) {
+                it.take(input.options.selectionLimit)
+              } else {
+                it
+              }
+            }
+
+            ImagePickerContractResult.Success(results)
+          } else {
+            if (intent.data != null) {
+              intent.data?.let { uri ->
+                val type = uri.toMediaType(contentResolver)
+                ImagePickerContractResult.Success(listOf(type to uri))
+              }
             } else {
-              it
+              uris.firstOrNull()?.let { uri ->
+                val type = uri.toMediaType(contentResolver)
+                ImagePickerContractResult.Success(listOf(type to uri))
+              } ?: ImagePickerContractResult.Error
             }
           }
-
-          ImagePickerContractResult.Success(results)
-        } else {
-          if (intent.data != null) {
-            intent.data?.let { uri ->
-              val type = uri.toMediaType(contentResolver)
-              ImagePickerContractResult.Success(listOf(type to uri))
-            }
-          } else {
-            uris.firstOrNull()?.let { uri ->
-              val type = uri.toMediaType(contentResolver)
-              ImagePickerContractResult.Success(listOf(type to uri))
-            } ?: ImagePickerContractResult.Error
-          }
-        }
-      } ?: ImagePickerContractResult.Error
+        } ?: ImagePickerContractResult.Error
+      }
+    } catch (e: Exception) {
+      Log.e("ExpoImagePicker", "Image library result handling failed", e)
+      ImagePickerContractResult.Error
+    }
